### PR TITLE
Updated to work with Kodi 18.

### PR DESCRIPTION
**Pull request by glk1001**
1) Many changes to Main.cpp to handle modern GL
2) Added multi-threaded workers to speed up goom calculations while rendering
3) Removed Goom II fly-in motif
4) Minor tweaks to old Goom code

### CODE REVIEW

**Comment by notspiff** [comment]
sorry for forgetting about this. please rebase.

**Comment by glk1001** [comment]
Hi, rebased and force pushed.

**Comment by AlwinEsch** [comment]
To fix the build and allow on more OS you need to add glm to depends, see as example https://github.com/xbmc/visualization.fishbmc/tree/master/depends

File: lib/kodi/utils/Time.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#pragma once
```
Comment (AlwinEsch) [inline lib/kodi/utils/Time.h]: Don't add the Time.h, thought on begin to use it on Kodi's addon headers but with the new C++11 functionality not needed anymore.

**Comment by AlwinEsch** [comment]
Have tested on my Linux and has worked fine :smile:, Thanks!

What is the exact reason of the ./test folder, to do independent test e.g. for other versions of Goom? If yes better then also add a CMakeLists.txt for a independent build and a "NOTE" or "README.md" to know why added.

**Comment by AlwinEsch** [comment]
I make you currently changes to your request code to support also GLES and cleanup. Comes later a request to your git.

**Comment by AlwinEsch** [comment]
@glk1001 Have added request to your repo, but seen also there is already another from @notspiff related to the needed depends.

https://github.com/glk1001/visualization.goom/pull/2

**Comment by glk1001** [comment]
Sorry, not much experience with pull-requests so I apologize if I've screwed up merges. Removed test directory - only good for initial testing.

**Comment by AlwinEsch** [comment]
Thanks a lot for your rework, is now in. With the git history is not the nicest but does not matter for me :smirk:

Repo also moved now to xbmc itself to have more access and support to this addon.
Since jenkins builds it now, I will still thoroughly test the individual OS and if everything is OK, I will create a release.

**Pull request merged**